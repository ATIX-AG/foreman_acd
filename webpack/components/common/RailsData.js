import React from 'react';
import PropTypes from 'prop-types';

const RailsData = ({
  view,
  parameter,
  value,
}) => {
  const id = "foreman_acd_"+ view +"_"+ parameter;
  const name = "foreman_acd_"+ view +"["+ parameter +"]";

  return (
    <input
      id={id}
      name={name}
      value={value}
      type="hidden"
    />
  );
}

RailsData.propTypes = {
  view: PropTypes.string.isRequired,
  parameter: PropTypes.string.isRequired,
};

export default RailsData;
