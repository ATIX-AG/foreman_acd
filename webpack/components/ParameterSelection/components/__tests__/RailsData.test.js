import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import RailsData from '../RailsData';

const noop = () => {};

const fixtures = {
  'should render rails data input field': {
    view: 'nice_view',
    parameter: 'nice_param',
    value: 'nice_value',
  },
};

describe('RailsData', () =>
  testComponentSnapshotsWithFixtures(RailsData, fixtures));
