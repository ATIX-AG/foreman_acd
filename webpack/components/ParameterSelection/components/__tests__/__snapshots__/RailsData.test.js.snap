// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RailsData should render rails data input field 1`] = `
<input
  id="foreman_acd_nice_view_nice_param"
  name="foreman_acd_nice_view[nice_param]"
  type="hidden"
  value="nice_value"
/>
`;
